tests/terraform/checks/resource/gcp/test_GoogleBigQueryDatasetPublicACL.py

```diff
@@ -51,6 +51,83 @@ def test_success(self):
+                         "friendly_name": ["test"],
+                         "description": ["This is a test description"],
+                         "location": ["EU"],
+                         "default_table_expiration_ms": [3600000],
+                         "access": [{"role": ["EDITOR"], "domain": ["example.com"]}]
+                         }
+        scan_result = check.scan_resource_conf(conf=resource_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
+
+    def test_success_view(self):
+        resource_conf = {"dataset_id": ["example_dataset"],
+                         "friendly_name": ["test"],
+                         "description": ["This is a test description"],
+                         "location": ["EU"],
+                         "default_table_expiration_ms": [3600000],
+                         "access": [{"view": {"projectId": "foo", "datasetId": "bar", "tableId": "buzz"}}]
+                         }
+        scan_result = check.scan_resource_conf(conf=resource_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
+
+    def test_success_routine(self):
+        resource_conf = {"dataset_id": ["example_dataset"],
+                         "friendly_name": ["test"],
+                         "description": ["This is a test description"],
+                         "location": ["EU"],
+                         "default_table_expiration_ms": [3600000],
+                         "access": [{"routine": {"projectId": "foo", "datasetId": "bar", "routineId": "buzz"}}]
+                         }
+        scan_result = check.scan_resource_conf(conf=resource_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
+
+    def test_success_dataset(self):
+        resource_conf = {"dataset_id": ["example_dataset"],
+                         "friendly_name": ["test"],
+                         "description": ["This is a test description"],
+                         "location": ["EU"],
+                         "default_table_expiration_ms": [3600000],
+                         "access": [{"dataset": {"dataset": {"datasetId": "foo","projectId": "bar"},"targetTypes": ["VIEWS"]}}]
+                         }
+        scan_result = check.scan_resource_conf(conf=resource_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
+
+    def test_success_iam_member(self):
+        resource_conf = {"dataset_id": ["example_dataset"],
+                         "friendly_name": ["test"],
+                         "description": ["This is a test description"],
+                         "location": ["EU"],
+                         "default_table_expiration_ms": [3600000],
+                         "access": [{"role": ["READER"], "iam_member": ["deleted:serviceAccount:foo@bar.iam.gserviceaccount.com?uid=1234567890"]}]
+                         }
+        scan_result = check.scan_resource_conf(conf=resource_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
```

Comment:
This happens when you delete service account.
Existing service account is appear with user_by_email.